common/src/main/java/io/pravega/common/util/AsyncIterator.java

```diff
@@ -76,4 +78,25 @@
             return result;
         };
     }
+
+    /**
+     * Processes the remaining elements in the AsyncIterator until the specified {@link Predicate} returns false.
+     *
+     * @param until A Predicate that will be used to the decide the number of elements on which the Consumer will be invoked.
```

Comment:
The predicate does not decide the number of elements. It returns true if the collection can continue, and false otherwise.

Reply:
updated java dock